packages/fx-core/src/component/generator/utils.ts

```diff
@@ -235,3 +239,97 @@ export function zipFolder(folderPath: string): AdmZip {
   zip.addLocalFolder(folderPath);
   return zip;
 }
+
+export async function downloadDirectory(
+  sampleUrl: string,
+  dstPath: string,
+  concurrencyLimits = sampleConcurrencyLimits,
+  retryLimits = sampleDefaultRetryLimits
+): Promise<void> {
+  const urlInfo = parseSampleUrl(sampleUrl);
+  const { samplePaths, fileUrlPrefix } = await getSampleFileInfo(urlInfo, retryLimits);
+  await downloadSampleFiles(
+    fileUrlPrefix,
+    samplePaths,
+    dstPath,
+    urlInfo.dir,
+    retryLimits,
+    concurrencyLimits
+  );
+}
+
+type SampleUrlInfo = {
+  owner: string;
+  repository: string;
+  ref: string;
+  dir: string;
+};
+function parseSampleUrl(url: string): SampleUrlInfo {
```

Comment:
It is better to have a blank line between these two method. I wonder why this is checked by eslint/prettier?